### DISCUSSION TITLE
Question RE Dynamic Filtering with Lookdown

### DISCUSSION THREAD

**Discussion by wenstarner** [khoait/DCE.PCF#133] [Q&A] [answered]
Hello,

I'm using Lookdown on two fields, Category and Sub-Category.

I have Lookdown working on Category. With Sub-Category, I'm trying to get the values to change based on the value in Category.

I have two challenges. I am not sure if they are bugs.

1. Dynamic filtering with a custom filter seems to work in the Sub-Category field based on the value in the Category field only when I use {{ logical_name_on_form**_.name_** }} - it doesn't seem to work without ".name"

2. Also, the Sub-Category lookup only gets updated to limit the dropdown items after selecting the Category value AND refreshing the form. It does not seem to work immediately after selecting the value for Category. 

Is there something I'm missing here?

**Comment by khoait** [comment]
Hi @wenstarner good find on no.1, when the field on form is a lookup field, you can access its `.name` like you did or even `.id` if you want. This feature is similar in the PolyLookup control, but not documented well in Lookdown wiki. I'll update it.
For no.2, the Subcategory filter should reflect the change in Category right away without saving and refreshing the form. I'll check again.

**Reply by khoait** [reply]
Ok, this is not as simple as expected. I'll need to figure out how to react to dependent fields change 🤔

**Reply by wenstarner** [reply]
Thanks!

If it helps, LookupDropdown works for me with dependent fields (though I like your component better!):
https://pcf.gallery/lookup-dropdown-pcf/

Maybe the source for that will yield a clue on how to implement?

Cheers

**Reply by khoait** [reply]
Thanks! I'll have a look in there.

**Reply by khoait** [reply, marked as answer]
hi @wenstarner, glad to share that I've figured it out and fixed it here #134. 

the LookupDropdown you shared only has one dependent lookup as a bound field while Lookdown can support any number of dependent fields through dynamic template, so there's not much I can refer from there.

this fix will be included in the next minor release.

**Reply by wenstarner** [reply]
That’s amazing news—thanks, Khao! And noted RE your component supporting many fields with a dynamic template. As I mentioned, I much prefer your component!

Thanks again for your time and effort in making this all available to the community. I am a huge fan of yours!

**Reply by khoait** [reply]
thanks for your inputs and feedback! They really help to motivate me and improve the components. cheers!
